src/data/move.ts

```diff
@@ -262,6 +258,18 @@ export abstract class Move implements Localizable {
     return false;
   }
 
+  isAttackMove(): this is AttackMove {
+    return false;
+  }
```

Comment:
```suggestion
  isAttackMove(): this is AttackMove {
    return this.category === MoveCategory.PHYSICAL || this.category === MoveCategory.SPECIAL;
  }
```
The other type guards can also probably pull from existing move fields in some way.

Reply:
This would remove the need to override them in the subclasses, right?